development/image-url-helper/pkg/list/imagelister.go

```diff
@@ -123,70 +123,76 @@ func GetWalkFunc(resourcesDirectory string, images, testImages *ImageList, image
 		}
-		if !ImageListContains(*testImages, testImage) {
-			*testImages = append(*testImages, testImage)
-		}
+		testImages[testImage.FullImageURL()] = testImage
 		components[testImage.FullImageURL()] = append(components[testImage.FullImageURL()], component)
 	}
 }
 
-// RemoveDoubles removes all duplicates
-func RemoveDoubles(images ImageList) ImageList {
-	var dedupedImages ImageList
-	for _, image := range images {
-		exists := false
-		for _, deduped := range dedupedImages {
-			if image == deduped {
-				exists = true
+// // RemoveDoubles removes all duplicates
+// func RemoveDoubles(images ImageList) ImageMap {
+// 	var dedupedImages ImageList
+// 	for _, image := range images {
+// 		exists := false
+// 		for _, deduped := range dedupedImages {
+// 			if image == deduped {
+// 				exists = true
+// 			}
+// 		}
+// 		if !exists {
+// 			dedupedImages = append(dedupedImages, image)
+// 		}
+// 	}
+// 	return dedupedImages
+// }
+
+// GetInconsistentImages returns a list of images with the same URl but different versions or hashes
+func GetInconsistentImages(images ImageMap) ImageMap {
+	inconsistent := make(ImageMap)
+
+	for imageName, image := range images {
+		for image2Name, image2 := range images {
+			if image.ImageURL() == image2.ImageURL() {
+				inconsistent[imageName] = image
+				inconsistent[image2Name] = image2
 			}
 		}
-		if !exists {
-			dedupedImages = append(dedupedImages, image)
-		}
 	}
-	return dedupedImages
-}
 
-// GetInconsistentImages returns a list of images with the same URl but different versions or hashes
-func GetInconsistentImages(images ImageList) ImageList {
-	var inconsistent ImageList
-	hasDoubles := make(map[string]ImageList)
+	return inconsistent
+}
 
+// PrintImages prints otu list of images and their usage in components
+func PrintImages(images ImageMap, imageComponentsMap ImageToComponents) {
+	imageNames := make([]string, 0)
 	for _, image := range images {
-		hasDoubles[image.ImageURL()] = append(hasDoubles[image.ImageURL()], image)
+		imageNames = append(imageNames, image.FullImageURL())
 	}
+	sort.Strings(imageNames)
 
-	for _, images := range hasDoubles {
-		if len(images) > 1 {
-			inconsistent = append(inconsistent, images...)
-		}
+	for _, fullImageURL := range imageNames {
+		components := imageComponentsMap[fullImageURL]
+		fmt.Printf("%s, used by %s\n", fullImageURL, strings.Join(components, ", "))
 	}
-
-	return inconsistent
 }
 
-// PrintImages prints otu list of images and their usage in components
-func PrintImages(images ImageList, imageComponentsMap ImageToComponents) {
-	sort.Sort(images)
-	for _, image := range images {
-		components := imageComponentsMap[image.FullImageURL()]
-		fmt.Printf("%s, used by %s\n", image.FullImageURL(), strings.Join(components, ", "))
+// MergeImageMap merges images map into target one
+func MergeImageMap(target ImageMap, source ImageMap) {
+	for key, val := range source {
+		target[key] = val
```

Comment:
You can add simple check if key exist in target map and skip overwriting value if key is already present. Less writes.

```suggestion
if _, ok := target[key]; !ok {
    target[key] = val
    }
```